### ISSUE TITLE
AVT 1 - React Dropdown has 2 DAP violations

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#2827]
## Environment
Tested on macOS Mojave 10.14.5, Version 74.0.3729.157 (Official Build) (64-bit), with VoiceOver

## Description
DAP found 2 violations (see screenshot)
![Screen Shot 2019-05-23 at 2 13 05 PM](https://user-images.githubusercontent.com/21676914/58280311-f0679580-7d65-11e9-818a-d5b58463ba3b.png)
Axe-Core found the same issues (see screenshot)
![Screen Shot 2019-05-23 at 2 18 12 PM](https://user-images.githubusercontent.com/21676914/58280337-f78ea380-7d65-11e9-9d8d-26c51de058b1.png)

**Comment by emyarod** [comment]
just for clarification, are the `aria-owns` and `aria-controls` attributes only necessary when the dropdown is open? I am assuming that the error is because when the dropdown is closed, the menu is not rendered so the attributes are referring to something that does not exist

**Comment by elizabethsjudd** [comment]
@emyarod they are necessary as long as the menu is in the DOM despite the state of the menu. Is it dynamically created every time it's opened or simply created when it's opened for the first time?  If it's the first time only, you should add the attribute then and leave it, otherwise you'll need to add/remove the attribute each time.  However, I have not tested this with a screen reader where the list doesn't exist on component creation/load so screen reader users may be missing information that there is an element controlled by the button when they first interact with it.

**Comment by emyarod** [comment]
@elizabethsjudd yeah it's added and removed from the DOM, I opened #3139 to toggle the attributes based on whether or not the menu is present

**Comment by elizabethsjudd** [comment]
@emyarod thanks for the clarification as to the behavior and creating the issue! I would definitely have screen readers users (team able) test it as like I said I'm not sure if not getting that information from the button when you first interact with it would be confusing to their workflow.  I'm experienced with screen reader user but definitely wouldn't say i'm an expert at them so getting someone who relies on AT would be a really good perspective.

**Comment by snidersd** [comment]
After retesting this issue DAP still reports 1 of the 2 violations (see screenshot):
<img width="1303" alt="Screen Shot 2019-09-27 at 2 04 00 PM" src="https://user-images.githubusercontent.com/21676914/65795472-f31d3180-e12f-11e9-92ce-e0f36112be5f.png">

**Comment by snidersd** [comment]
@dakahn In addition to the above DAP violation the Dropdown with items as components also has the following violations when the Dropdown is expanded:
<img width="1158" alt="Screen Shot 2019-10-03 at 8 58 28 AM" src="https://user-images.githubusercontent.com/21676914/66133466-51c52e00-e5bc-11e9-9e5f-e7b46bfbfd4a.png">
